File: Client/src/app/shared/models/order.ts
import { Address } from "./user";

export interface OrderToCreate{
    basketID : string;
    deliveryMethod : number;
    shipToAddress : Address
}

export interface Order{
    id: number;
    buyerEmail : string;
    orderDate : string;
    shipToAddress : Address;
    deliveryMethod : string;
    shippingPrice : number;
    orderItems : OrderItem[];
    subtotal : number;
    status : string;
    total : number
}

export interface OrderItem{
    productID : number;
    productName : string;
    imgUrl : string;
    price : number;
    quantity : number;
}

export interface OrderCreated{
    id : number
}

File: API/Core/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public Order()
        {
            
        }
        public Order(IReadOnlyList<OrderItem> orderItems, string buyerEmail, Address shipToAddress, DeliveryMethod deliveryMethod, float subtotal,string paymentIntentId)
        {
            OrderItems = orderItems;
            BuyerEmail = buyerEmail;
            ShipToAddress = shipToAddress;
            DeliveryMethod = deliveryMethod;
            Subtotal = subtotal;
            PaymentIntentID = paymentIntentId;
        }
        public string BuyerEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public Address ShipToAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public float Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public string? PaymentIntentID { get; set; }

        public float GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }

    }
}
